Fixes #233 by implementing support for triggered demand in in the SubscriberBlackboxVerification

viktorklang opened the pull request:
Fixes #233 @purplefox & @ktoso?

ktoso commented on tck/src/main/java/org/reactivestreams/tck/SubscriberBlackboxVerification.java:
```diff
@@ -48,6 +48,16 @@ protected SubscriberBlackboxVerification(TestEnvironment env) {
+   * By default this method does nothing.
+   */
+  public void triggerRequest(final Subscriber<? super T> subscriber) {
+
```
"this <strike>page</strike> method body was intentionally left blank." :wink:

ktoso commented on tck/src/test/java/org/reactivestreams/tck/SyncTriggeredDemandSubscriberTest.java:
```diff
@@ -0,0 +1,48 @@
+      }
+
+      @Override public void onComplete() {
+        System.out.println("Accumulated: " + acc);
```
remove before merge?

viktorklang replied:
:+1:

ktoso commented on tck/src/test/java/org/reactivestreams/tck/support/SyncTriggeredDemandSubscriber.java:
```diff
@@ -0,0 +1,117 @@
+/**
+ * SyncSubscriber is an implementation of Reactive Streams `Subscriber`,
+ * it runs synchronously (on the Publisher's thread) and requests one element
+ * at a time and invokes a user-defined method to process each element.
```
Add a small notice about why/how it's a "triggered" one.

viktorklang replied:
:+1:

ktoso commented on tck/src/main/java/org/reactivestreams/tck/SubscriberBlackboxVerification.java:
```diff
@@ -74,6 +84,7 @@ public void required_spec201_blackbox_mustSignalDemandViaSubscriptionRequest() t
     blackboxSubscriberTest(new BlackboxTestStageTestRun() {
       @Override
       public void run(BlackboxTestStage stage) throws InterruptedException {
+        triggerRequest(stage.subProxy().sub());
```
looks good

viktorklang replied:
Thanks Kon!

ktoso commented on tck/src/main/java/org/reactivestreams/tck/SubscriberBlackboxVerification.java:
```diff
@@ -248,6 +259,7 @@ public void run(BlackboxTestStage stage) throws Throwable {
         final BlackboxSubscriberProxy<T> probe = stage.createBlackboxSubscriberProxy(env, sub);
 
         pub.subscribe(probe);
+        triggerRequest(sub);
```
:+1:

ktoso commented on tck/src/test/java/org/reactivestreams/tck/support/SyncTriggeredDemandSubscriber.java:
```diff
@@ -0,0 +1,117 @@
+    }
+  }
+
+  public boolean triggerDemand(final long n) {
```
add comments before merge here please

viktorklang replied:
:+1:

ktoso commented:
This looks v. good I think, thanks Viktor!

viktorklang commented:
@ktoso @purplefox & @reactive-streams/contributors Addressed review feedback in the following commit: https://github.com/reactive-streams/reactive-streams-jvm/commit/658c95eaed8d47de1076eeaac0abb7e12e767140

viktorklang commented:
@reactive-streams/contributors & @purplefox This PR is ready to merge, Tim please verify that this fixes your issue and contributors-please raise any objections you might have (this is completely without impact on already TCK-passing implementations). I'll merge this on the 30th of March unless there are objections. Thanks

purplefox commented:
This works for me.

(Aside: One small issue, not specifically related to this PR, I had to install Java 7 to build the TCK as it doesn't build using Java 8 - small JavaDoc issue. Java 7 will be EOL in a month)

viktorklang commented:
Great that it worked. What was the Java Doc issue?

-- 
Cheers,
√
On 28 Mar 2015 08:51, "Tim Fox" <notifications@github.com> wrote:

> This works for me.
>
> (Aside: One small issue, not specifically related to this PR, I had to
> install Java 7 to build the TCK as it doesn't build using Java 8 - small
> JavaDoc issue. Java 7 will be EOL in a month)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reactive-streams/reactive-streams-jvm/pull/252#issuecomment-87188435>
> .
>

viktorklang commented:
@purplefox Alright, I found and fixed the issue, it was due to the requirement to qualify nested classes in Javadoc links.
